overlord/snapstate/snapmgr.go

```diff
@@ -551,6 +556,101 @@ func (m *SnapManager) ensureUbuntuCoreTransition() error {
 	return nil
 }
 
+// cleanupSnapConfineApparmor will remove all but the current apparmor
+// profiles for the snap-confine binary from core
+func (m *SnapManager) cleanupSnapConfineApparmor(currentSnapConfineProfilePath string) error {
+	apparmorProfilePathPattern := strings.Replace(filepath.Join(dirs.SnapMountDir, "/core/*/usr/lib/snapd/snap-confine"), "/", ".", -1)[1:]
+
+	glob, err := filepath.Glob(filepath.Join(dirs.SystemApparmorDir, apparmorProfilePathPattern))
+	if err != nil {
+		return err
+	}
+
+	for _, path := range glob {
+		if path == currentSnapConfineProfilePath {
+			continue
+		}
+
+		// not using apparmor.UnloadProfile() because it uses a
+		// different cachedir
+		if output, err := exec.Command("apparmor_parser", "-R", filepath.Base(path)).CombinedOutput(); err != nil {
+			logger.Noticef("cannot remove apparmor %s: %q", path, output)
+		}
+		if err := os.Remove(path); err != nil && !os.IsNotExist(err) {
+			return err
+		}
+		if err := os.Remove(filepath.Join(dirs.SystemApparmorCacheDir, filepath.Base(path))); err != nil && !os.IsNotExist(err) {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (m *SnapManager) addSnapConfineApparmor(snapConfineInCore, apparmorProfilePath string) error {
+	root := filepath.Join(dirs.SnapMountDir, "/core/current/")
+
+	// FIXME: make this more generic once we start supporting more
+	//        distros with apparmor around snap-confine, i.e.
+	//        ship `snap-confine.apparmor.in` and use that as the base
+	//        because core is ubuntu and classic host might be anything
+	apparmorProfile, err := ioutil.ReadFile(filepath.Join(root, "/etc/apparmor.d/usr.lib.snapd.snap-confine"))
```

Comment:
This is a location in classic, which means it depends on the packaging, right?  At the top of this PR we have a `snapConfinePath` variable being used which assumes this location is not hardcoded.

Same concern below and perhaps elsewhere. Need to watch out for the difference between a path in core, which we're sure about, and in the outer system, which we're not.